### PR TITLE
expose service to outside, fix #140

### LINKED ISSUE #140: Creating Routes for Services
OpenShift can expose Services using Routes.

It would be nice if Kompose created default Routes for some services.

User can specify for which services Route should be created for example like this:

```
kompose convert --expose-service=<service name>
```

cc: @kbsingh

### PR DESCRIPTION
**Pull request by containscafeine**

### CODE REVIEW

**Comment by cdrage** [comment]
i think this needs rebasing, you've got some README.md changes in here.

**Comment by containscafeine** [comment]
@cdrage yeah, it's a WIP right now, things are going to get worse before it gets any better :)

File: pkg/transformer/openshift/openshift.go [hunk 1]
```diff
@@ -160,6 +184,10 @@ func (o *OpenShift) Transform(komposeObject kobject.KomposeObject, opt kobject.C
 			objects = append(objects, svc)
 		}
 
+		if service.ExposeService != "" {
+			objects = append(objects, o.initRoute(name, service))
+		}
```
Comment (surajssd) [inline pkg/transformer/openshift/openshift.go]: Also we can expose a service only if we can create a service in first place. So this if block should be inside `if o.PortsExist(name, service)`.

Comment (rtnpro) [inline pkg/transformer/openshift/openshift.go]: :+1:

**Comment by rtnpro** [comment]
@containscafeine I tested the example you have created and it seems to work correctly. I also had a look at the code. It looks good, overall.

File: pkg/transformer/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -152,6 +152,45 @@ func (k *Kubernetes) InitDS(name string, service kobject.ServiceConfig) *extensi
...
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if service.ExposeService != "true" {
```
Comment (surajssd) [inline pkg/transformer/kubernetes/kubernetes.go]: is there a way that we enforce user to provide either "true" or a valid url?

Comment (surajssd) [inline pkg/transformer/kubernetes/kubernetes.go]: i am asking because what if user provides value as `false`

**Comment by surajssd** [comment]
@containscafeine please send docs with this, we don't want user to miss out on this cool feature due to lack of docs

File: pkg/transformer/kubernetes/kubernetes.go (hunk 2 above)
Comment (containscafeine) [inline pkg/transformer/kubernetes/kubernetes.go]: In that case, `false` gets treated as a hostname. If the user does not want to expose, he would rather not write (or comment) this in the compose file.

Even I am divided on this. Should we enforce, (true, false, hostname), or just (true, hostname)?

Comment (surajssd) [inline pkg/transformer/kubernetes/kubernetes.go]: okay np :)

Comment (containscafeine) [inline pkg/transformer/kubernetes/kubernetes.go]: So went ahead with  (true, hostname)

**Comment by cdrage** [comment]
@containscafeine tests seem to fail since `glide-vc --only-code --no-tests` needs to be ran again in the directory for vendoring. missing one odd file.

**Comment by cdrage** [comment]
commits need to be squashed and then LGTM from me :)

**Comment by containscafeine** [comment]
@cdrage @surajssd - added docs, squashed the commits, refactored the ingress unit test since last review. Should be good to go now.

**Comment by cdrage** [comment]
@containscafeine just conflicting right now :)

**Comment by containscafeine** [comment]
@cdrage fixed

**Comment by surajssd** [comment]
LGTM!

**Pull request merged**